lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
I'd in favour of excluding NetCDF from callbacks.  All our known use cases are to support GRIB2 and PP saving

If I want to save an attribute to a NetCDF file, I add it to my cube: done. 

It's when there's a translation process taking place that I want to be able to post patch, to make my PP/GRIB field conform to a particular pattern which Iris may not understand, or may be ambiguous, or poorly represented in CF semantics